### PR TITLE
[browser] Marshalling of resolved Tasks/promises is not resolved in JS/C#

### LINKED ISSUE #104772: javascript calling a method with return Task type in .net 9.0 preview6, the returned promise is always in pending state
### Description

return Task<T> is fine.

### Reproduction Steps

# c#
```
[JSExport]
internal static Task ReturnTask()
{
    return Task.CompletedTask;
}
```

# javascript
```
let p = (await getAssemblyExports(assemblyName)).namespaceName.className.ReturnTask();
console.log(p);
```

### Expected behavior

PromiseState is resolved

### Actual behavior

PromiseState is pending

### Regression?

Yes, .net8.0 is fine.

### Known Workarounds

_No response_

### Configuration

.NET SDK：9.0.100-preview.6.24328.19
Microsoft.NETCore.App: 9.0.0-preview.6.24327.7

### Other information

_No response_

### PR DESCRIPTION
**Pull request by ilonatommy**
The PR solves the problem with void Task marshalling. Fixes https://github.com/dotnet/runtime/issues/104772.

- Added lib test for https://github.com/dotnet/runtime/issues/104772
- C# to JS in net8.0.7 returned `Promise is resolved. Success.` -> regression
- Added lib test for `JSImport` that is the same as the issue but with opposite marshalling direction.

cc @pavelsavara

Might be connected: https://github.com/dotnet/runtime/pull/95411

### CODE REVIEW

**Comment by pavelsavara** [comment]
WBT is too heavy weight integration test for this. Let's add test to `src\libraries\System.Runtime.InteropServices.JavaScript\tests\System.Runtime.InteropServices.JavaScript.UnitTests\System\Runtime\InteropServices\JavaScript\JSImportTest.cs` instead

File: src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/JavaScriptTestHelper.mjs [hunk 1]
```diff
@@ -246,6 +246,22 @@ export function invoke2(arg1, name) {
...
 }
 
+export function returnResolvedPromise() {
+    return Promise.resolve();
+}
+
+export async function invokeReturnCompletedTask() {
+    const JavaScriptTestHelper = dllExports.System.Runtime.InteropServices.JavaScript.Tests.JavaScriptTestHelper;
+    const p = JavaScriptTestHelper['ReturnCompletedTask'];
+    Promise.race([p, delay(100)])
```
Comment (pavelsavara) [inline src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/JavaScriptTestHelper.mjs:256]: `await Promise.race`

Comment (ilonatommy) [reply src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/JavaScriptTestHelper.mjs]: Right, it should be there. I don't see any change in behavior, though

Comment (ilonatommy) [reply src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/JavaScriptTestHelper.mjs]: it's `()`! I save function as `p` but never execute it. Sorry, the commit is flying in a sec

**Comment by ilonatommy** [comment]
[ST](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-105464-merge-6258cedd19164cee9c/WasmTestOnV8-ST-System.Runtime.InteropServices.JavaScript.Tests/1/console.5d66ccaa.log?helixlogtype=result)

[MT JSImport](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-105464-merge-0338cd438d404adba3/WasmTestOnChrome-MT-System.Runtime.InteropServices.JavaScript.Tests/1/console.cebcacea.log?helixlogtype=result)

MT JSExport - disabled(?)

**Review by pavelsavara — APPROVED** [review]

**Pull request merged**